src/control/cmd/dmg/pool_test.go

```diff
@@ -19,21 +19,116 @@ import (
 
 	"github.com/dustin/go-humanize"
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/pkg/errors"
 
+	"github.com/daos-stack/daos/src/control/cmd/dmg/pretty"
 	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
 	"github.com/daos-stack/daos/src/control/common/test"
-	. "github.com/daos-stack/daos/src/control/common/test"
 	"github.com/daos-stack/daos/src/control/lib/control"
 	"github.com/daos-stack/daos/src/control/lib/daos"
 	"github.com/daos-stack/daos/src/control/lib/ranklist"
 	"github.com/daos-stack/daos/src/control/logging"
 	"github.com/daos-stack/daos/src/control/system"
 )
 
-var (
-	defaultPoolUUID = MockUUID()
-)
+func Test_Dmg_PoolTierRatioFlag(t *testing.T) {
+	for name, tc := range map[string]struct {
+		input     string
+		expRatios []float64
+		expString string
+		expErr    error
+	}{
+		"empty": {
+			expErr: errors.New("no tier ratio specified"),
+		},
+		"less than 100%": {
+			input:  "10,80",
+			expErr: errors.New("must add up to"),
+		},
+		"total more than 100%": {
+			input:  "30,90",
+			expErr: errors.New("must add up to"),
+		},
+		"non-numeric": {
+			input:  "0.3,foo",
+			expErr: errors.New("invalid"),
+		},
+		"negative adds up to 100": {
+			input:  "-30,130",
+			expErr: errors.New("0-100"),
+		},
+		"0,0": {
+			input:  "0,0",
+			expErr: errors.New("must add up to"),
+		},
+		"%,%": {
+			input:  "%,%",
+			expErr: errors.New("invalid"),
+		},
+		"defaults": {
+			input:     "defaults",
+			expRatios: defaultTierRatios,
+			expString: func() string {
+				rStrs := make([]string, len(defaultTierRatios))
+				for i, ratio := range defaultTierRatios {
+					rStrs[i] = pretty.PrintTierRatio(ratio)
+				}
+				return strings.Join(rStrs, ",")
+			}(),
+		},
+		"0": {
+			input:     "0",
+			expRatios: []float64{0, 1},
+			expString: "0.00%,100.00%",
+		},
+		"100": {
+			input:     "100",
+			expRatios: []float64{1},
+			expString: "100.00%",
+		},
+		"single tier padded": {
+			input:     "45.3",
+			expRatios: []float64{0.453, 0.547},
+			expString: "45.30%,54.70%",
+		},
+		"valid two tiers": {
+			input:     "0.3,99.7",
+			expRatios: []float64{0.003, 0.997},
+			expString: "0.30%,99.70%",
+		},
+		"valid three tiers": {
```

Comment:
For future support of level 3 storage ?

Reply:
Maybe? More just verifying that the type doesn't impose arbitrary constraints on the number of tiers. Those constraints (if any) would be the responsibility of the application code, not the input handling code.